internal/credentials/credentialls.go

```diff
@@ -0,0 +1,176 @@
+		if !exists {
+			r.Log.Warn("failed to find password in secret, generating new one")
+		} else {
+			passwordHash, err := bcrypt.GenerateFromPassword(password, bcrypt.DefaultCost)
+			if err != nil {
+				return "", fmt.Errorf("failed to generate password hash: %w", err)
+			}
+			return string(passwordHash), nil
+		}
+	} else if !errors.IsNotFound(err) {
+		return "", fmt.Errorf("failed to get kubeadmin password secret: %w", err)
+	}
+	r.Log.Infof("Generating admin password")
+	password, err := generateKubeadminPassword()
+	if err != nil {
+		return "", fmt.Errorf("failed to generate password: %w", err)
+	}
+	data := map[string][]byte{
+		"username": []byte(DefaultUser),
+		"password": []byte(password),
+	}
+
+	if err := r.createOrUpdateClusterCredentialSecret(ctx, cd, KubeadminPasswordSecretName(cd.Name), data, kubeadmincreds); err != nil {
+		return "", fmt.Errorf("failed to create kubeadmin password secret: %w", err)
+	}
+	passwordHash, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return "", fmt.Errorf("failed to generate password hash: %w", err)
+	}
+	return string(passwordHash), nil
+}
+
+func (r *Credentials) createOrUpdateClusterCredentialSecret(ctx context.Context, cd *hivev1.ClusterDeployment, name string, data map[string][]byte, secretType string) error {
+	secret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: cd.Namespace,
+		},
+		Data: data,
+	}
+
+	secret.Labels = AddLabel(secret.Labels, "hive.openshift.io/cluster-deployment-name", cd.Name)
+	secret.Labels = AddLabel(secret.Labels, "hive.openshift.io/secret-type", secretType)
+
+	deploymentGVK, err := apiutil.GVKForObject(cd, r.Scheme)
+	if err != nil {
+		r.Log.WithError(err).Errorf("error getting GVK for clusterdeployment")
+		return err
+	}
+	secret.OwnerReferences = []metav1.OwnerReference{{
+		APIVersion:         deploymentGVK.GroupVersion().String(),
+		Kind:               deploymentGVK.Kind,
+		Name:               cd.Name,
+		UID:                cd.UID,
+		BlockOwnerDeletion: pointer.Bool(true),
+	}}
+	mutateFn := func() error {
+		// Update the Secret object with the desired data
+		secret.Data = data
+		return nil
+	}
+	op, err := controllerutil.CreateOrUpdate(ctx, r.Client, secret, mutateFn)
+	if err != nil {
+		return fmt.Errorf("failed to create secret: %w", err)
+	}
+	r.Log.Infof("%s secret %s", secretType, op)
+	return nil
+}
+
+func KubeconfigSecretName(clusterDeploymentName string) string {
+	return clusterDeploymentName + "-admin-kubeconfig"
+}
+
+func KubeadminPasswordSecretName(clusterDeploymentName string) string {
+	return clusterDeploymentName + "-admin-password"
+}
+
+func generateKubeadminPassword() (string, error) {
+	const (
+		lowerLetters = "abcdefghijkmnopqrstuvwxyz"
+		upperLetters = "ABCDEFGHIJKLMNPQRSTUVWXYZ"
+		digits       = "23456789"
```

Comment:
Missing 1 and 0?
Was that intentional?

Reply:
Copied from the installer, looked at the same code in hypershift no 1 and 0 there as well.
:man_shrugging: